src/libmongoc/src/mongoc/mongoc-crypt.c

```diff
@@ -945,11 +948,44 @@ _mongoc_crypt_new (const bson_t *kms_providers,
       }
    }
 
+   if (!bypass_auto_encryption) {
+      mongocrypt_setopt_append_csfle_search_path (crypt->handle, "$SYSTEM");
+      if (!_crypt_check_error (crypt->handle, error, false)) {
+         goto fail;
+      }
+
+      if (csfle_override_path != NULL) {
+         mongocrypt_setopt_set_csfle_lib_path_override (crypt->handle,
+                                                        csfle_override_path);
+         if (!_crypt_check_error (crypt->handle, error, false)) {
+            goto fail;
+         }
+      }
+   }
+
    if (!mongocrypt_init (crypt->handle)) {
       _crypt_check_error (crypt->handle, error, true);
       goto fail;
    }
 
+   if (csfle_required) {
+      const char *s = mongocrypt_csfle_version_string (crypt->handle, NULL);
+      if (!s || strlen (s) == 0) {
```

Comment:
```suggestion
      uint32_t len = 0u;
      const char *s = mongocrypt_csfle_version_string (crypt->handle, &len);
      if (!s || len == 0u) {
```
Minor suggestion to avoid `strlen ()`.

Reply:
I wrote the API and didn't even recognize this. Oops.